src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -165,13 +167,29 @@ private static String getHash(@CheckForNull SCMRevision revision) {
     @Extension
     public static class JobCheckOutListener extends SCMListener {
 
+        /**
+         * Marks when the first Git checkout has being completed.
+         * When building PRs with merge strategy two checkouts are done, one to obtain the Jenkinsfile and one to obtain the source code.
+         * When importing libraries, each library also executes a checkout.
+         * This avoids sending multiple notifications to Bitbucket by remembering the first checkout.
+         */
+        private static class FirstCheckoutCompletedAction extends InvisibleAction {}
```

Comment:
You’ll need to keep track of what repo and commit hash as multiple checkouts of different repos will miss notifications otherwise.

Reply:
I thought that the build object was going to be exclusive to this build being run, but are you saying if there are builds executing in parallel, for the same job, this can mess with each other? I will test this now...

Reply:
I have tested starting 9 builds at the same time with and without my change, for the same job. So far it seems to work. 
Without my change each build shows 4 items on Bitbucket PR activity (one In Progress for PR with merge strategy, one  In Progress for checkout, one  In Progress for the global lib and one success). 
After my change it shows 2 each build (one for In Progress one for Success).
I am going to test now building different projects simultaneously.

Reply:
I have now tested with multiple projects building several times simultaneously and it works as well.

The object I am using `Run<?, ?> build` seems to be a different object for each build run, as I am printing it to logs and I get a different instance and toString() every-time. The toString is `system-team-groovy-demo/PR-35 #41`, `system-team-groovy-demo/PR-35 #42`, `system-team-groovy-demo/PR-35 #43` etc...

Since its a different object for each build, it does not affect each others action list and the if/else works correctly. It does not seems to be missing any notification.

This is the list. Its out of order by it always get one In Progress for each build and it never miss one:
<img width="449" alt="screen shot 2018-09-22 at 13 29 18" src="https://user-images.githubusercontent.com/4479787/45919448-8a8bdc80-be6b-11e8-9c7b-b9745f0dfbf6.png">

PS: Screenshot is not complete as there are more scrollable content above the screen.

Unless I am missing something it seems to work...

Reply:
@stephenc ^

Reply:
Multiple checkouts of different repos in the same run.

Can get there by:

* shared pipeline library 
* using `resolveScm` like in https://gitbox.apache.org/repos/asf?p=maven.git;a=blob;f=Jenkinsfile;h=9fa73cd31bbf6296a2c6d696c0f72137b4176790;hb=refs/heads/master#l62

You need the checkout to be initiated through the SCMSource codepath in order to cause the notifications to take place